test/bitarray.jl

```diff
@@ -203,6 +203,15 @@ timesofar("utils")
         @test_throws DimensionMismatch BitMatrix((isodd(i) for i in 1:3))
     end
 
+    @testset "constructor from infinite iterable" begin
+        struct InfiniteIter; state::Bool; end
+        Base.iterate(i::InfiniteIter, state=1) = (state, !state)
```

Comment:
```suggestion
        Base.iterate(i::InfiniteIter, state=true) = (state, !state)
```
Using `1` (an `Int`) as argument will not work:
```
julia> !1
ERROR: MethodError: no method matching !(::Int64)
```